cmake/list_insert_options.cmake

```diff
@@ -0,0 +1,55 @@
+#
+# Append key[=value] elements from one list to another list if they are not already in the target list.
```

Comment:
Since you are proposing the feature I will leave it up to you to decide if it is a warning by default or an error.

From my point of view this new feature will likely not be widely usable for e.g. setting the compiler standard version (which was one reason it was proposed in the first place I think). Simply because any situation where packages don't agree on the value are going to be a problem.